wicamo96/FieldScout
# Scouting

# Description

Added CRUD functionality for the scouting report functionality.

## Type of change


# Testing Instructions

### Create database and populate data if necessary
- If you haven't created the FieldScout database, please do so in visual studio community.
- Use this to fill out tables and columns: [SQL data](https://github.com/wicamo96/FieldScout/blob/main/SQL/01_FieldScout_Create_DB.sql)
- Use this to seed the database: [SQL Seed Data](https://github.com/wicamo96/FieldScout/blob/main/SQL/02_FieldScout_Seed_Data.sql)

### Testing Greenhouse Management CRUD
- Click the execute with debugger solid green button to launch the backend in visual studio community.
- From the client directory in a git terminal, run `npm run dev`.
- Once the Vite + React window pops up, you should be directed to the log in page.
- Log in with the default email.
- Navigate to the scouting menu through the navbar, you should see a list of houses that exist within your facility
- Select House A to view the bays within the house and to begin inputting data
- Select the Add Data button next to bay 1, when the modal appears, input data for each pest in each bay division and click Add
- If you click the Add Data button again you should see a message telling you data already exists and only have the option to cancel
- Click the edit button, when the modal appears, all of your previous selections should be present in the dropdowns. Edit the selections and click the Submit Edit button
- Check that your edits are reflected in the edit modal after submitting.
- Click the delete button and confirm delete
- When you click the edit or delete buttons after deleting the data, the messaging in the modals should reflect that there is no data and you aren't given the option to edit or delete any info


# Checklist:
